### ISSUE TITLE
List markers overlap margin of inline elements

### ISSUE THREAD

**Issue by danierutu** [squidfunk/mkdocs-material#2554]
- [x] I've read the [contribution guidelines][1] and agree with them

__I've found a bug and checked that ...__

- [x] ... the problem doesn't occur with the default MkDocs template
- [x] ... the problem is not in any of my customizations (CSS, JS, template)
- [x] ... the documentation does not mention anything about my problem
- [x] ... there are no open or closed issues that are related to my problem

## Description

When you add an inline admonition followed by a list (ordered or not), the list markers (numbers or bullets) are not affected by the margin of the admonition. Instead, it only applies to the `<li>` element.

I am pretty sure that this bug was introduced with 7.1.1 release, since I downgraded `mkdocs-material` to 7.1.0 and it was still fine. Users of the product I write docs for noticed the issue, and it turned out that the deployed version used 7.1.1 whereas the version in my Docker environment was still in 7.1.0.

The page that was affected also had some inline pictures, which used the `.inline` CSS class, so they were also affected by this bug.

### Expected behavior

From local 7.1.0

![image](https://user-images.githubusercontent.com/63101447/114447388-a5adcb80-9bd2-11eb-9df9-502f5af17aa0.png)
![image](https://user-images.githubusercontent.com/63101447/114447478-c2e29a00-9bd2-11eb-83e4-551cc361a163.png)

### Actual behavior

From deployed 7.1.1

![image](https://user-images.githubusercontent.com/63101447/114447723-f9b8b000-9bd2-11eb-94c9-5d6c44c04197.png)
![image](https://user-images.githubusercontent.com/63101447/114447637-e6a5e000-9bd2-11eb-8827-11e7742185bd.png)

### Steps to reproduce the bug

1. Upgrade to **7.1.1**.
2. Use a set-up like the one below.
3. Serve the site, the bug exists.
4. Downgrade to **7.1.0**.
5. Serve the site, the bug no longer exists.

```
!!! info inline "Inline admonition"
    This is some text.

- This is a list.
- This is a list.
- This is a list.
- This is a list.
```

### Package versions

* Python: 3.9.1
* MkDocs: 1.1.2
* Material: 7.1.0 (unaffected), 7.1.1 (affected)

### System information

* OS: Windows 21H1
* Browser: Microsoft Edge (reported from Firefox too)

  [1]: https://github.com/squidfunk/mkdocs-material/blob/master/CONTRIBUTING.md

**Comment by squidfunk** [comment]
Thanks for reporting. Related change in https://github.com/squidfunk/mkdocs-material/commit/c011da1fa4a615a0e57ca94e4c5d3a29e3bb7a06 which fixed #2501 but introduced this problem.

**Comment by squidfunk** [comment]
Potential fix in 21b543a03. I believe it's safe to add back `flow-root` to `ul` and `ol` elements, as my testing shows that the bottom margin compounding reported in #2501 does not occur with them.

**Comment by squidfunk** [comment]
Released as part of 7.1.2.
